packages/core/src/renderer/components/+namespaces/subnamespace-badge.tsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import styles from "./subnamespace-badge.module.scss";
+
+import React from "react";
+import type { Namespace } from "../../../common/k8s-api/endpoints";
+import { Tooltip } from "../tooltip";
+
+interface SubnamespaceBadgeProps extends React.HTMLAttributes<HTMLSpanElement> {
+  namespace: Namespace;
+  id: string;
+}
+
+export function SubnamespaceBadge({ namespace, id, ...other }: SubnamespaceBadgeProps) {
+  if (!namespace.isSubnamespace()) {
+    return null;
+  }
+
+  return (
+    <>
+      <span
+        className={styles.subnamespaceBadge}
+        data-testid={`subnamespace-badge-for-${namespace.getId()}`}
+        id={id}
+        {...other}
+      >
+        S
+      </span>
+      <Tooltip targetId={id}>
+        Subnamespace
```

Comment:
Looks like useless info (Subnamespace) for every namespace..
Why do we even need this component at all? I would define all these details directly in `<NamespaceTreeView/>`.
Also I don't like `if(!namespace.isSubnamespace()) return;` which looks like limit usages of a whole component in general.

Reply:
> Why do we even need this component at all?

To separate it from `NamespaceTreeView` and its styles as well. To reach Single Responsibility principle a bit more.

> Also I don't like if(!namespace.isSubnamespace()) return; which looks like limit usages of a whole component in general.

Moved this check to parent component.